chutney/action-impl/src/test/java/com/chutneytesting/action/kafka/KafkaBasicConsumeActionIntegrationTest.java

```diff
@@ -68,92 +82,204 @@ public void before() {
         actionExecutionResult = consumeAction.execute();
 
         assertThat(actionExecutionResult.status).isEqualTo(Failure);
 
-        // third time with reset
+        // Third time with reset
         Action consumeActionWithReset = getKafkaBasicConsumeAction(targetBuilder.build(), props, true);
         actionExecutionResult = consumeActionWithReset.execute();
 
         assertThat(actionExecutionResult.status).isEqualTo(Success);
         body = assertActionOutputsSize(actionExecutionResult, 1);
         assertThat(body.get(0).get("payload")).isEqualTo("1");
 
-        // third time without reset
+        // Third time without reset
         consumeAction = getKafkaBasicConsumeAction(targetBuilder.build(), props, false);
         actionExecutionResult = consumeAction.execute();
 
         assertThat(actionExecutionResult.status).isEqualTo(Failure);
+    }
 
+    @Nested
+    @DisplayName("Publish then consume binary message")
+    class PublishThenConsume {
+        @Test
+        @DisplayName("simple")
+        public void publish_and_consume_message_as_byte_array() {
+            // Given
+            TestTarget kafkaTarget = targetBuilder
+                .withProperty(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getCanonicalName())
+                .withProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getCanonicalName())
+                .withProperty(ConsumerConfig.GROUP_ID_CONFIG, GROUP)
+                .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, OffsetResetStrategy.EARLIEST.name().toLowerCase())
+                .build();
+
+            byte[] payload = "Hello la France !!".getBytes();
+            var publishAction = new KafkaBasicPublishAction(
+                kafkaTarget, uniqueTopic, null, payload, null, null, logger
+            );
+            var publishResult = publishAction.execute();
+            assertThat(publishResult.status).isEqualTo(Success);
+
+            // When
+            var consumeAction = new KafkaBasicConsumeAction(
+                kafkaTarget, uniqueTopic, GROUP, emptyMap(), 1, null, null,
+                APPLICATION_OCTET_STREAM.getMimeType(), "3 s", null, false, logger
+            );
+            var actionExecutionResult = consumeAction.execute();
+
+            // Then
+            assertThat(actionExecutionResult.status).isEqualTo(Success);
+            List<Map<String, Object>> body = assertActionOutputsSize(actionExecutionResult, 1);
+            assertThat(body.get(0).get("payload")).isInstanceOf(byte[].class);
+            assertThat(actionExecutionResult.outputs.get("payloads")).asList().first()
+                .isInstanceOf(byte[].class)
+                .isEqualTo(payload);
+        }
+
+        @Test
+        @DisplayName("header selector")
+        public void publish_and_consume_message_as_byte_array_with_header_selector() {
+            // Given
```

Comment:
there are nothing specific to byte in header selector. it should be not in binary section